woocommerce/woocommerce-android
# Feature/ai product creation survey

Closes: #10330 


### Description

Display a bottom sheet in product detail after the user creates 3 products using AI product generation feature. 

### Testing instructions

1. Log into a site eligible for product creation with AI
2. Create 2 new products using AI OR apply this patch: 
```diff
Subject: [PATCH] Fix minor UI glitch for dark mode
---
Index: WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductDetailViewModel.kt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductDetailViewModel.kt b/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductDetailViewModel.kt
--- a/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductDetailViewModel.kt	(revision 538f8e5600020625cdfd1d1775b1d56e3ea80046)
+++ b/WooCommerce/src/main/kotlin/com/woocommerce/android/ui/products/ProductDetailViewModel.kt	(date 1702396720010)
@@ -363,8 +363,8 @@
         }
     }
 
-    private fun shouldShowAIProductCreationSurvey() = navArgs.isAIContent &&
-        appPrefsWrapper.numberOfProductsCreatedUsingAi == MINUM_NUMBER_OF_AI_CREATED_PRODUCTS_TO_SHOW_SURVEY
+    private fun shouldShowAIProductCreationSurvey() = navArgs.isAIContent
+//        appPrefsWrapper.numberOfProductsCreatedUsingAi == MINUM_NUMBER_OF_AI_CREATED_PRODUCTS_TO_SHOW_SURVEY
 
     private fun startAddNewProduct() {
         val defaultProduct = createDefaultProductForAddFlow()

```
3. Create another product using AI feature. After tapping on "Save product as draft" check the the survey bottom sheet is displayed when landing on product detail screen. 
4. Click on "Start the Survey" and check the survey is loaded correctly. 
5. Check the following events were tracked for the given interaction: 
`product_creation_ai_survey_confirmation_view_displayed` and `product_creation_ai_survey_start_survey_button_tapped`
7. Navigate back and check the bottom sheet is gone. 

### Images/gif


https://github.com/woocommerce/woocommerce-android/assets/2663464/2e4471ed-0457-425e-b856-c2b6e370377c
